### PR TITLE
Cleanup and upgrade tests, linter configs, CI

### PR DESCRIPTION
**Pull request by dapirian**

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -30,21 +32,18 @@ jobs:
     strategy:
       matrix:
         linter-version: [KnownGoodVersion, Latest]
+
     steps:
       - name: Checkout
         uses: actions/checkout@v3
 
-      - name: Locate trunk
```
Comment (TylerJang27) [inline .github/workflows/pr.yaml:37]: This is necessary (or something similar) to make sure trunk is installed on the CI machine

Comment (dapirian) [reply .github/workflows/pr.yaml:37]: See below, I changed it to use the trunk launcher via npm

**Review by TylerJang27 — APPROVED** [review]
Lgtm.

Couple process questions:
1. Were you using an autoformatter or just going heuristically in sorting the trunk.yaml action fields?
2. Any explicit rationale for removing cspell?

**Comment by dapirian** [comment]
> Lgtm.
> 
> Couple process questions:
> 
> 1. Were you using an autoformatter or just going heuristically in sorting the trunk.yaml action fields?
> 2. Any explicit rationale for removing cspell?

1. Manual - with the intention to build opinionated good auto-sorting into trunk at some point of trunk.yaml

**Pull request merged**